src/highlight_manager.ts

```diff
@@ -126,3 +146,49 @@ export class HighlightManager implements Disposable {
         disposeAll(this.disposables);
     }
 }
+
+class PendingHLUpdates {
+    // maps from grid number to a set of functions that indicate whether or not a grid should update
+    private pendingGridUpdates: Map<number, (() => boolean)[]>;
+
+    constructor() {
+        this.pendingGridUpdates = new Map();
+    }
+
+    size(): number {
+        return this.pendingGridUpdates.size;
+    }
+
+    entries(): [number, () => boolean][] {
+        return Array.from(this.pendingGridUpdates.entries()).map(([key, checks]) => {
+            const anyValid = () => this.evaluateChecks(checks);
```

Comment:
```suggestion
          const anyValid = () => checks.some(check => check());
```

Reply:
This actually is subtly incorrect. `.some()` will stop evaluation at the first item that returns `true`, which we don't want here, as we want to process all the elements to calculate the highlights. 

I only know this because it was my first iteration 😅

Reply:
need a (1-line) code comment that explains that

Reply:
Sure. And I'll ensure a test validates it too